src/components/PopupWithForm.js

```diff
@@ -16,6 +18,14 @@ export default class PopupWithForm extends Popup {
     return inputValue;
   }
 
+  setLoadingState(isLoading) {
+    if (this._submitButton) {
+      this._submitButton.textContent = isLoading
+        ? this._defaultButtonText
+        : "Saving...";
+    }
+  }
```

Comment:
**COULD BE IMPROVED**

You can make a special method for getting a form element from `PopupWithForm` not to **search** it in `index.js`



```js

  getForm() {
    return this._popupForm
  }


```